File: 2903-insert-greatest-common-divisors-in-linked-list/insert-greatest-common-divisors-in-linked-list.cpp
class Solution {
public:
    ListNode* insertGreatestCommonDivisors(ListNode* head) {
         ListNode *root=head;
         while(root->next){
             if(root->next){
                 int val=__gcd(root->val,root->next->val);
                 ListNode *tem=new ListNode(val);
                 tem->next=root->next;
                 root->next=tem;
                 
             }
             root=root->next->next;
         }
         return head;
    }
};

File: 2903-insert-greatest-common-divisors-in-linked-list/insert-greatest-common-divisors-in-linked-list.py
import math
class Solution:
    def insertGreatestCommonDivisors(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or head.next==None:
            return head
        current=head
        while current and current.next:
            x=math.gcd(current.val,current.next.val)
            new_node=ListNode(x)
            new_node.next=current.next
            current.next=new_node
            current=current.next.next
        return head
        